lib/ruby_lsp/server.rb

```diff
@@ -315,13 +315,14 @@ def text_document_did_open(message)
           language_id: language_id,
         )
 
-        if document.source.length > Requests::SemanticHighlighting::MAXIMUM_CHARACTERS_FOR_HIGHLIGHT
+        if document.source.length > MAXIMUM_CHARACTERS_FOR_EXPENSIVE_FEATURES
           send_message(
             Notification.new(
               method: "window/showMessage",
               params: Interface::ShowMessageParams.new(
                 type: Constant::MessageType::WARNING,
-                message: "This file is too long. For performance reasons, semantic highlighting will be disabled",
+                message: "This file is too long. For performance reasons, semantic highlighting and " \
```

Comment:
What if we just say "For performance reasons, some Ruby LSP features will be disabled". That can then cover any aspect that we might want to disable in future.

Reply:
It's not a strong opinion, but I would rather be specific about what we're turning off so that users know what exactly what they can expect to be working. If we're too generic, people won't know if something is a bug or if it's intentionally disabled.